robots/pkg/circuitbreaker/circuitbreaker.go

```diff
@@ -69,10 +69,17 @@ func (g *CircuitBreaker) WrapRetryableFunc(retryableFunc func() error) func() er
 func (g *CircuitBreaker) isOpenAndNotFeasibleForRetry() (bool, error) {
 	g.mutex.RLock()
 	defer g.mutex.RUnlock()
-	return g.open && !g.blockedUntil.Before(time.Now()), g.lastErr
+	return g.open && g.isRetryBlocked(), g.lastErr
+}
+
+func (g *CircuitBreaker) isRetryBlocked() bool {
+	return !g.blockedUntil.IsZero() && !time.Now().After(g.blockedUntil)
 }
 
 func (g *CircuitBreaker) updateState(err error) {
+	if g.isRetryBlocked() {
+		return
+	}
```

Comment:
IIUC what you are seeing is a read without a lock, could be fixed by adding a `RLock/RUnlock` call to `isRetryBlocked`. WDYT?

Reply:
I prefer to make this specific read under the `Lock` context. Yes, it is a read, but it is done to allow or disallow a write.
Let me explain: if two process want to update, they both read and this can be done in parallel, so both are allowed to write. This is not a big deal since the subsequent write will be atomic (due to the `Lock`), but this does not prevent having multiple (very near) write, that could be avoided. WDYT?

Reply:
Yes, that's a good point! Thanks!